addons/note/models/note_note.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from openerp import api, fields, models
+from openerp.tools import html2plaintext
+
+class NoteNote(models.Model):
+    """ Note """
+    _name = 'note.note'
+    _inherit = ['mail.thread']
+    _description = "Note"
+    _order = 'sequence'
+
+    #read the first line (convert html into text)
+    @api.one
+    @api.depends('memo')
+    def _get_note_first_line(self):
+        self.name = (self.memo and html2plaintext(self.memo) or "").strip().replace('*', '').split("\n")[0]
+
+    # return the default stage for the uid user
+    def _get_default_stage_id(self):
+        return self.env['note.stage'].search([('user_id', '=', self.env.uid)], limit=1).id or False
+
+    def _set_stage_per_user(self):
+        for record in self:
+            if record.stage_id:
+                record.stage_ids = record.stage_id + record.stage_ids.filtered(lambda stage: stage.user_id != self.env.user)
```

Comment:
1/ I don't understand this code, and it seems weird
2/ Use |
3/ What the hell is the purpose of this code ?